### ISSUE TITLE
Feature Request: Improve error formatting

### ISSUE THREAD

**Issue by ShariJoosten** [CrowdStrike/gofalcon#112]
Hello!

While invoking the function client.client.RealTimeResponseAdmin.RTRCreateScripts we sometimes get a 403 Forbidden response. The error message we then get is: `*real_time_response_admin.RTRCreateScriptsForbidden · [POST /real-time-response/entities/scripts/v1][403] rTRCreateScriptsForbidden &{Errors:[0xc0008a1460] Meta:0xc000628a80}` As you can see, having pointers being printed to string isn't really helpful here. At https://github.com/CrowdStrike/gofalcon/blob/7614b9dff6ab4e12c3dd0dcf211e23b6edeae7f4/falcon/client/real_time_response_admin/r_t_r_create_scripts_responses.go#L213 we can see that the pointers are coming from `o.Payload`, which refers to the field Payload in RTRCreateScriptsForbidden https://github.com/CrowdStrike/gofalcon/blob/7614b9dff6ab4e12c3dd0dcf211e23b6edeae7f4/falcon/client/real_time_response_admin/r_t_r_create_scripts_responses.go#L209 which is a pointer to an instance of MsaReplyMetaOnly https://github.com/CrowdStrike/gofalcon/blob/7614b9dff6ab4e12c3dd0dcf211e23b6edeae7f4/falcon/models/msa_reply_meta_only.go#L21-L29 MsaReplyMetaOnly in return again has two fields: `Errors` is an array of pointers to `MsaAPIError` and `Meta` is an pointer to `MsaMetaInfo`. If we'd add the following code to the [MsaAPIError object](https://github.com/CrowdStrike/gofalcon/blob/7614b9dff6ab4e12c3dd0dcf211e23b6edeae7f4/falcon/models/msa_api_error.go)

```
func (m *MsaAPIError) String() string {
	return fmt.Sprintf("%+v", *m)
}
```

and if we'd add the following code to the [MsaMetaInfo object](https://github.com/CrowdStrike/gofalcon/blob/7614b9dff6ab4e12c3dd0dcf211e23b6edeae7f4/falcon/models/msa_meta_info.go)

```
func (m *MsaMetaInfo) String() string {
	return fmt.Sprintf("%+v", *m)
}
```

Then I'd stop seeing pointers and instead see the values in the error messages the library returns. I'd normally open up a pull request myself but on each file on line 1 I see https://github.com/CrowdStrike/gofalcon/blob/7614b9dff6ab4e12c3dd0dcf211e23b6edeae7f4/falcon/models/msa_api_error.go#L1 So that's why I now open up an issue instead :P

Update:
Debugging another bug I also find https://github.com/CrowdStrike/gofalcon/blob/7614b9dff6ab4e12c3dd0dcf211e23b6edeae7f4/falcon/models/domain_device_details_response_swagger.go#L21-L34 where we run into the same problem, this time with the `[GET /devices/entities/devices/v1][400] GetDeviceDetails default &{Errors:[0xc00086f060] Meta:0xc0008d7dd0 Resources:[]}` function call. Luckily the suggested solution would also work here, since the models are reused. I do however notice the new field called `Resources`. Currently I'm not running into pointers here yet because the list is empty, but because Resources is also an array of pointers, I'd again end up with an array of printed pointers if I had any resources. The solution here again would be to overwrite the string method of the pointer of the `DomainDeviceSwagger` model, as we did with the other two already.

**Comment by isimluk** [comment]
Hello @ShariJoosten,

Thanks for the issue report! 

Your analysis is very accurate and I like the solution you have proposed. I have created pr #116 for this.

What I have noticed, however, is that I cannot simply introduce: 
```
func (m *MsaAPIError) String() string {
	return fmt.Sprintf("%+v", *m)
}
```

as that would end-up with outputs like:
```
&{Errors:[{Code:0xc0004605ac Message: 0xc000126360}]
```
(tested with go1.16.5).

So, I have introduced slightly more verbose methods to do the formatting.

----
Note: There is actually way to modify the behaviour of autogenerated code by introducing new files into the source directory. Only those files that are autogenerated are being re-generated periodically, the others are left untouched.

**Comment by ShariJoosten** [comment]
I haven't had the time to fully dive into how these files were being autogenerated and how to workaround this. Luckily you did :D Thanks for showing me how it's done!

Yes, I've failed to notice that the `Code` and `Message` fields were once again pointers. I've looked at your commit and I see you nicely worked around this. I've added two comments about unconventional spaces.

Thanks for your help and I'm looking forward to a new release with improved error formatting :D
